cloudaux/orchestration/aws/s3.py

```diff
@@ -17,14 +17,57 @@
 from cloudaux.orchestration import modify
 
 from botocore.exceptions import ClientError
+from bunch import Bunch
 import logging
 import json
 
 
 logger = logging.getLogger('cloudaux')
 
 
-def get_grants(bucket_name, include_owner=False, **conn):
+FLAGS=Bunch(
+    GRANTS=1,
+    GRANT_REFERENCES=1,
+    OWNER=1,
+    LIFECYCLE=2,
+    LOGGING=4,
+    POLICY=8,
+    TAGS=16,
+    VERSIONING=32,
+    WEBSITE=64,
+    CORS=128,
+    NOTIFICATIONS=256,
+    ACCELERATION=512,
+    REPLICATION=1024,
+    ANALYTICS=2048,
+    METRICS=4096,
+    INVENTORY=8192,
+    CREATED_DATE=16384,
+    ALL=32767)
+
+
+class FlagRegistry:
+    r = list()
+
+    @classmethod
+    def register(cls, flag, key):
+        def decorator(fn):
+            flag_list = flag
+            key_list = key
+            if type(flag) not in [list, tuple]:
+                flag_list = [flag] 
+            if type(key) not in [list, tuple]: 
+                key_list = [key]
+            for idx in xrange(len(flag_list)):
+                cls.r.append(dict(flag=flag_list[idx], key=key_list[idx], method=fn, rtv_ix=idx))
```

Comment:
good point - I initially wrote the FlagRegistry class to try and handle the `get_grants()` method, which is unique in that it has multiple return values that should get mapped to different fields in the return dictionary.

```
+# @FlagRegistry.register(
 +#     flag=(FLAGS.GRANTS, FLAGS.GRANT_REFERENCES, FLAGS.OWNER),
 +#     key=('grants', 'grant_references', 'owner'))
 +def get_grants(bucket_name, include_owner=True, **conn):
```

The `rtv_ix` was suppose to indicate the index of the return value.  However, I kind of abandoned the idea of having FlagRegistry support these more complex interactions.  I don't want to call the method three times.